README.md

```diff
@@ -50,31 +56,35 @@ export default connect(
 
 ## API
 
-Adds the following methods to the redux store
+Adds the following methods to the redux store.
 
 #### `store.selectIpfsReady()`
 
-- `boolean` - Is the IPFS instance ready to use yet?
+- `boolean` - Is the IPFS instance ready to use yet?.
 
 #### `store.selectIpfsInitFailed()`
 
-- `boolean` - Did the IPFS instance fail to start?
+- `boolean` - Did the IPFS instance fail to start?.
 
 #### `store.selectIpfsIdentity()`
 
-- `Object` - The last resolved value of `ipfs.id()`
+- `Object` - The last resolved value of `ipfs.id()`.
 
-#### `store.selectIpfsApiAddress()`
+#### `store.selectIpfsProvider()`
 
-- `string` - The API address of the IPFS instance.
+- `string` - Can be `window.ipfs`, `js-ipfs-api` or `js-ipfs`.
+
+#### `store.selectIpfsApiOpts()`
+
+- `string` - The API options of the IPFS instance.
 
 #### `store.doInitIpfs()`
 
 - Create a new IPFS instance. This will `window.ipfs` if you have [IPFS Companion](https://github.com/ipfs-shipyard/ipfs-companion) installed, or a [`js-ipfs-api`](https://github.com/ipfs/js-ipfs-api) instance otherwise.
 
-### `store.doUpdateIpfsAPIAddress(address)`
+### `store.doUpdateIpfsApiOpts(opts)`
 
-- Updates the API Address to `address`.
+- Updates the API Options to `opts`.
```

Comment:
Add a warning here that calling this will dispose of any existing ipfs instance and will only try to init an js-ipfs-api instance, and it won't try window.ipfs or js-ipfs.